components/workflow.rst

```diff
@@ -19,6 +19,46 @@ You can install the component in 2 different ways:
 * :doc:`Install it via Composer </components/using_components>` (``symfony/workflow`` on `Packagist`_);
 * Use the official Git repository (https://github.com/symfony/workflow).
 
-For more information, see the code in the Git Repository.
+.. include:: /components/require_autoload.rst.inc
+
+Creating a Workflow
+-------------------
+
+The workflow component gives you an object oriented way to define a process
+or a life cycle that your object goes through. Each step or stage in the
+process is called a *place*. You do also define *transitions* to that describes
+the action to get from one place to another.
+
+.. image:: /_images/components/workflow/states_transitions.png
+
+A set of places and transitions creates a **definition**. A workflow needs
+a ``Definition`` and a way to write the states to the objects, (i.e. an
+instance of a :class:`Symfony\\Component\\Workflow\\MarkingStore\\MarkingStoreInterface`.
+
+Consider the following example for a blog post. A post can have places:
+'draft', 'review', 'rejected', 'published'. You can define the workflow
+like this::
+
+    use Symfony\Component\Workflow\Definition;
+    use Symfony\Component\Workflow\Transition;
+    use Symfony\Component\Workflow\Workflow;
+    use Symfony\Component\Workflow\MarkingStore\ScalarMarkingStore;
+
+    $states = ['draft', 'review', 'rejected', 'published'];
+
+    // Define a transaction with a name, where to go from and where to go to
+    $transitions[] = new Transition('to_review', 'draft', 'review');
+    $transitions[] = new Transition('publish', 'review', 'published');
+    $transitions[] = new Transition('reject', 'review', 'rejected');
+
+    $definition = new Definition($states, $transitions);
+    $definition->setInitialPlace('draft');
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking);
+
+The ``Workflow`` can now help you to decide what actions that are allowed
+on a blog post depending on what *place* it is in. This will keep your domain
```

Comment:
> In this case, do we have to all the rules in BlogPost class for instance ?

No, this is out of scope of `BlogPost`. These rules are not naturally belong to the `BlogPost` itself; they are external. They have nothing to do with internal state of the `BlogPost`.

In your example, you have different [bounded contexts](http://martinfowler.com/bliki/BoundedContext.html). It's a tool for model decomposition. For example, you may have `Blog` context with all blog-related business logic. And then, let's say, you want to add some commercial stuff for specific partners: not more than 3 free posts per day via Facebook Application. But `Blog` context "works" like as usual, it may know nothing about commercial stuff and Facebook applications at all. You just add some overlying commercial context like `BlogCommercial` with limits, pricing, discounts, etc. Technically, we just ask `BlogCommercial` first in order to find out if user is allowed to post or not.